cloudtracker/cli.py

```diff
@@ -234,59 +256,78 @@ def main(principals, organization_id, account_id, credentials, principal_types,
+
+        cdx_logger.info("Checking S3 paths in logging account...")
         for log_level, cloudtrail_log_path in cloudtrail_log_paths.items():
 
             try:
+                cdx_logger.info(f"Attempting to create Boto3 session for logging account ({log_level} level check).")
+                logging_account_session = None
                 if creds['type'] == 'self':
-                    cloudtrail_log_paths[log_level]["boto3_session"] = boto3.Session(
+                    logging_account_session = boto3.Session(
                         aws_access_key_id=creds['aws_access_key_id'],
                         aws_secret_access_key=creds['aws_secret_access_key'],
                     )
-
                 elif creds['type'] == 'assumerole':
-                    cloudtrail_log_paths[log_level]["boto3_session"] = boto3.Session(
+                    logging_account_session = boto3.Session(
                         aws_access_key_id=creds['aws_access_key_id'],
                         aws_secret_access_key=creds['aws_secret_access_key'],
                         aws_session_token=creds['session_token'],
                         region_name=creds.get('primary_region', "us-east-1")
                     )
+                cloudtrail_log_paths[log_level]["boto3_session"] = logging_account_session
+                cdx_logger.info("Logging account Boto3 session created successfully.")
 
             except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
-                logging.debug("Error occurred calling boto3.Session().", exc_info=True)
-                logging.error(
+                cdx_logger.debug("Error occurred calling boto3.Session().", exc_info=True)
+                cdx_logger.error(
                     (
-                        "Unable to initialize the default AWS session, an error occurred: %s. Make sure your AWS credentials "
+                        "Unable to initialize AWS session for logging account ({log_level} level check), an error occurred: %s. Make sure your AWS credentials "
                         "are configured correctly, your AWS config file is valid, and your credentials have the SecurityAudit "
                         "policy attached."
                     ),
                     e,
                 )
                 continue
 
-            try:
-                s3 = cloudtrail_log_paths[log_level]["boto3_session"].client('s3')
-                s3.get_object(
-                    Bucket=logging_account["bucketName"],
-                    Key=cloudtrail_log_path["path"],
-                )
-                cloudtrail_log_paths[log_level]["present"] = True
-                cloudtrail_log_paths[log_level]["bucket"] = logging_account["bucketName"]
-            except (s3.exceptions.NoSuchKey, s3.exceptions.NoSuchBucket, botocore.exceptions.ClientError) as e:
-                continue
+            if cloudtrail_log_paths[log_level].get("boto3_session"):
```

Comment:
Remove this if else condition. if its fail to create session it will throw an error